src/dotnet/APIView/APIView/Model/TokenTreeModel.cs

```diff
@@ -0,0 +1,182 @@
+            token.RenderClasses.Add("keyword");
+            return token;
+        }
+
+        public static StructuredToken CreateKeywordToken(SyntaxKind syntaxKind)
+        {
+            return CreateKeywordToken(SyntaxFacts.GetText(syntaxKind));
+        }
+
+        public static StructuredToken CreateKeywordToken(Accessibility accessibility)
+        {
+            return CreateKeywordToken(SyntaxFacts.GetText(accessibility));
+        }
+
+        public static StructuredToken CreatePunctuationToken(string value)
+        {
+            var token = new StructuredToken(value);
+            token.RenderClasses.Add("punctuation");
+            return token;
+        }
+
+        public static StructuredToken CreatePunctuationToken(SyntaxKind syntaxKind)
+        {
+            return CreatePunctuationToken(SyntaxFacts.GetText(syntaxKind));
+        }
+
+        public static StructuredToken CreateTypeNameToken(string value)
+        {
+            var token = new StructuredToken(value);
+            token.RenderClasses.Add("type-name");
+            return token;
+        }
+
+        public static StructuredToken CreateMemberNameToken(string value)
+        {
+            var token = new StructuredToken(value);
+            token.RenderClasses.Add("member-name");
+            return token;
+        }
+
+        public static StructuredToken CreateLiteralToken(string value)
+        {
+            var token = new StructuredToken(value);
+            token.RenderClasses.Add("literal");
+            return token;
+        }
+
+        public static StructuredToken CreateStringLiteralToken(string value)
+        {
+            var token = new StructuredToken(value);
+            token.RenderClasses.Add("string-literal");
+            return token;
+        }
+    }
+
+    public class APITreeNode
+    {
+        public string Name { get; set; }
+        public string Id { get; set; }
+        public string Kind { get; set; }
+        public HashSet<string> Tags { get; set; } = new HashSet<string>();
+        public Dictionary<string, string> Properties { get; set; } = new Dictionary<string, string>();
+        public List<StructuredToken> TopTokens { get; set; } = new List<StructuredToken>();
+        public List<StructuredToken> BottomTokens { get; set; } = new List<StructuredToken>();
+        public List<APITreeNode> Children { get; set; } = new List<APITreeNode>();
+
+        public override int GetHashCode()
+        {
+            int hash = 17;
+            hash = hash * 23 + (Name != null ? Name.GetHashCode() : 0);
+            hash = hash * 23 + (Id != null ? Id.GetHashCode() : 0);
+            hash = hash * 23 + (Kind != null ? Kind.GetHashCode() : 0);
+            return hash;
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (obj == null || GetType() != obj.GetType())
+            {
+                return false;
+            }
+
+            var other = (APITreeNode)obj;
```

Comment:
```suggestion to reduce type check cost
            var other = obj as APITreeNode;
            if (other == null)
            {
                return false;
            }
```

Reply:
Your suggestion does not check if the passed type is APITreeNode.

Reply:
It is implicitly checked by `as` operator. If passed type is not APITreeNode then `other` will be null. That's how object is tried to cast using `as`. Operator `as` helps to reduce additional type check.